bundler.js

```diff
@@ -91,7 +91,16 @@ export async function installBundler(bundlerVersionInput, lockFile, platform, ru
   } else {
     const gem = path.join(rubyPrefix, 'bin', 'gem')
     const bundlerVersionConstraint = /^\d+\.\d+\.\d+/.test(bundlerVersion) ? bundlerVersion : `~> ${bundlerVersion}`
-    await exec.exec(gem, ['install', 'bundler', '-v', bundlerVersionConstraint])
+    const floatVersion = common.floatVersion(rubyVersion)
+    if ((bundlerVersionConstraint === '~> 2') && (floatVersion >= 2.3) && (floatVersion <= 2.5)) {
+      // stop warnings similar to
+      // Your RubyGems version (2.6.14.4)) has a bug that prevents `required_ruby_version` from working for Bundler
+      await exec.exec(gem, ['update', '--system', '--no-document'])
```

Comment:
We might need to do something about that one, not sure yet, but I think installing `Bundler < 2.3.0` is less risky: https://github.com/ruby/setup-ruby/issues/228#issuecomment-1009156811
Anyway, I think best to not change that in this PR

Reply:
Pre-compiled gems are probably more common with Windows, and the `required_ruby_version` bug has definitely caused problems.  So, I'd rather see the above than `Bundler < 2.3.0`.  There has been a lot of recent work in Bundler/RubyGems, so it would be good to install the most recent version available...

Regardless, I'll remove the `gem update`.

Reply:
I wonder if the required_ruby_version only affects RubyGems or also bundler, if only rubygems it seems a much smaller issue, anyway best discussed on a separate issue.

Reply:
I can't recall, but it's a PITA.  I think Bundler, but it uses code in RubyGems?  As I recall, it happens when a pre-compiled gem exists, but doesn't support the Ruby version, which will always happen with Ruby master/head.  But, it installs it anyway...